File: Python/Programmers/Level 2/큰 수 만들기/s2.py
# 출처: 프로그래머스 다른 사람의 풀이
# 뒤의 수와 비교하되, 제거해주고나서 이전에 봤던 숫자로 돌아가서 계속 비교한다.

def solution(number, k):
    i = 0
    while i < len(number) - 1 and k > 0:
        if number[i] < number[i + 1]:
            number = number[:i] + number[i + 1:]
            if i != 0:
                i -= 1  # 이 부분이 핵심
            k -= 1
        else:
            i += 1
    if k > 0:
        return number[:-k]
    return number

File: Python/Programmers/Level 2/이진 변환 반복하기/s2.java
class Solution {
    public int[] solution(String s) {
        int cnt = 0;
        int zero = 0;
        while (!s.equals("1")) {
            long one = s.chars().filter(ch -> ch == '1').count();
            zero += s.length() - one;
            s = Integer.toBinaryString((int) one);
            cnt += 1;
        }
        int[] answer = {cnt, zero};
        return answer;
    }
}